src/wallet/mod.rs

```diff
@@ -196,24 +197,55 @@ pub enum AddressIndex {
     Reset(u32),
 }
 
+/// A derived address and the index it was found at
+/// For convenience this automatically derefs to `Address`
+#[derive(Debug, PartialEq)]
+pub struct AddressInfo {
+    /// Child index of this address
+    pub index: u32,
+    /// Address
+    pub address: Address,
+}
+
+impl Deref for AddressInfo {
+    type Target = Address;
+
+    fn deref(&self) -> &Self::Target {
+        &self.address
+    }
+}
```

Comment:
Just stumbled over this while upgrading to bdk 0.8.0. Not sure if it has been mentioned in the discussion but `Deref` is really only meant for smart pointers according to this documentation: https://doc.rust-lang.org/std/ops/trait.Deref.html

Would you accept a PR that removes this `impl` block again?

---

A separate discussion is on how to make access to the address in the struct more convenient. An alternative would be to make `get_new_address` return a tuple instead which can be instantly deconstructed at the call-site without having to import and name a type (`AddressInfo` in this case):

```rust
let (address, index) = wallet.get_new_address().unwrap();
```

or this if you don't care about the index:

```rust
let (address, _) = wallet.get_new_address().unwrap();
```

It is a question of taste really. The tuple approach looses you some documentation because the returned integer is effectively unnamed and `u32` itself doesn't say much about what it is.

Reply:
I'm not so sure about this, I think there are a few other cases where we use `Deref` in the code that are not smart pointers. If there aren't any real downsides to using them I would just keep this as is

Reply:
for reference, here https://github.com/pretzelhammer/rust-blog/blob/master/posts/tour-of-rusts-standard-library-traits.md#deref--derefmut there is some explanation of why `Deref` should be avoided for things that are not smart pointers

Reply:
That's a good point, but I don't think we are abusing it in this case. We are not really trying to reproduce inheritance like in other OOP langs, it's just a convenience that allows the use of methods like `script_pubkey()` directly on `AddressInfo`